File: 06.AndroidUsingJava/DesignPatterns/newMVP2/app/src/main/java/com/example/newmvp/MainActivity.java
package com.example.newmvp;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

import androidx.appcompat.app.AppCompatActivity;

import com.example.newmvp.allProdcutView.AllProductsActivityView;
import com.example.newmvp.favProductView.FavAcitivtyView;

public class MainActivity extends AppCompatActivity {

    Button btnGetAllProducts;
    Button btnGetFavProducts;
    Button btnExit;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        btnGetAllProducts = findViewById(R.id.btnGetAllProductsID);
        btnGetFavProducts = findViewById(R.id.btnGetFavProductID);
        btnExit = findViewById(R.id.btnExitID);
        btnGetAllProducts.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent outAllIntent = new Intent(MainActivity.this, AllProductsActivityView.class);
                startActivity(outAllIntent);
            }
        });

        btnGetFavProducts.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent outFavIntent = new Intent(MainActivity.this, FavAcitivtyView.class);
                startActivity(outFavIntent);
            }
        });
        btnExit.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                finish();
            }
        });



    }
}

File: AndroidUsingKotlin/Session_7/MVVM/app/src/main/java/com/example/mvvm/MainActivity.kt
package com.example.mvvm

import android.annotation.SuppressLint
import android.content.Intent
import android.os.Bundle
import android.widget.Button
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat
import com.example.mvvm.AllProducts.AllProducts
import com.example.mvvm.FavProducts.AllFavProducts

class MainActivity : AppCompatActivity() {

    private lateinit var  btnAllProducts: Button
    private lateinit var  btnFavProducts: Button
    private lateinit var  btnExit   : Button

    @SuppressLint("MissingInflatedId")
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        setContentView(R.layout.activity_main)

        btnAllProducts = findViewById(R.id.btnAllProductsID)
        btnFavProducts = findViewById(R.id.btnFavProductsID)
        btnExit = findViewById(R.id.btnExitID)

        btnAllProducts.setOnClickListener {
            val outIntent = Intent(this@MainActivity, AllProducts::class.java)
            startActivity(outIntent)
        }

        btnFavProducts.setOnClickListener {
            val outIntent = Intent(this@MainActivity, AllFavProducts::class.java)
            startActivity(outIntent)
        }

        btnExit.setOnClickListener {
               finish()
        }
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main)) { v, insets ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
            insets
        }
    }
}